lib/sql.js

```diff
@@ -601,8 +623,33 @@ SQLConnector.prototype.updateAttributes = function(model, id, data, options, cb)
  * @param {Function} cb The callback function
  * @returns {ParameterizedSQL} The UPDATE SQL statement
  */
-SQLConnector.prototype.buildUpdate = function(model, where, data, options) {
+SQLConnector.prototype._buildUpdate = function(model, where, data, options) {
   var fields = this.buildFieldsForUpdate(model, data);
+  return this._constructUpdateQuery(model, where, fields);
+};
+
+/**
+ * Build the UPDATE statement for replacing
+ * @param {String} model The model name
+ * @param {Object} where The where object
+ * @param {Object} data The data to be changed
+ * @param {Object} options The options object
+ * @param {Function} cb The callback function
+ * @returns {ParameterizedSQL} The UPDATE SQL statement for replacing fields
+ */
+SQLConnector.prototype._buildReplace = function(model, where, data, options) {
```

Comment:
The `build...` methods are extension points that a subclass can override to customize its implementations. It's better to have a consistent method signature so that the subclasses can have full access to the request.